pkg/controller/account/account_controller.go

```diff
@@ -333,28 +334,24 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 	}
 
 	if (currentAcctInstance.Status.State == "") && (currentAcctInstance.Status.Claimed == false) {
-		// before doing anything make sure we are not over the limit if we are just error
-		accountTotal, err := TotalAwsAccounts(awsSetupClient)
-		if err != nil {
-			reqLogger.Info("Failed to get AWS account total from AWS api", "Error", err.Error())
-			return reconcile.Result{}, err
-		}
-
-		if accountTotal >= awsLimit {
-			reqLogger.Error(ErrAwsAccountLimitExceeded, "AWS Account limit reached", "Account Total", accountTotal)
-			return reconcile.Result{}, ErrAwsAccountLimitExceeded
-		}
 
 		var awsAccountID string
 
 		if currentAcctInstance.Spec.AwsAccountID == "" {
+			// before doing anything make sure we are not over the limit if we are just error
+			if totalaccountwatcher.TotalAccountWatcher.Total >= AwsLimit {
+				reqLogger.Error(ErrAwsAccountLimitExceeded, "AWS Account limit reached", "Account Total", totalaccountwatcher.TotalAccountWatcher.Total)
+				return reconcile.Result{}, ErrAwsAccountLimitExceeded
+			}
 
 			// Build Aws Account
 			awsAccountID, err = r.BuildAccount(reqLogger, awsSetupClient, currentAcctInstance)
 			if err != nil {
 				return reconcile.Result{}, err
 			}
 
+			totalaccountwatcher.TotalAccountWatcher.Total += 1
```

Comment:
Just a quick though, want to make sure I understand. So since this var is from an instance of the totalAccountWatcher object that lives in the main, this value gets persisted through each reconcile right? Have you tested creating an account and then reconciling again, and the outcome? I assume this works but just want to make sure this actually writes to memory outside the scope of the reconcile function.